types/address.go

```diff
@@ -99,8 +99,16 @@ func AccAddressFromBech32(address string) (addr AccAddress, err error) {
 		return nil, err
 	}
 
-	if len(bz) != AddrLen {
-		return nil, errors.New("Incorrect address length")
+	validator := GetConfig().GetAddressValidator()
+	if validator != nil {
+		err = validator(bz)
```

Comment:
Do you have an example implementation of an address verifier that differs from the default logic? Would its sole contents just be a length check?

Reply:
One example would be to check addresses for different length constraints. For instance, maybe allowing 20 or 21 byte addresses instead of just 20 bytes.

It could also check that addresses have a certain prefix. For instance, maybe 21 byte addresses could require a prefix between 0-5 as the first byte to differentiate the type of key.